packages/graph-explorer/src/components/Panel.tsx

```diff
@@ -55,7 +55,7 @@ const PanelHeader = React.forwardRef<
   <div
     ref={ref}
     className={cn(
-      "bg-background-default flex min-h-[48px] w-full shrink-0 items-center gap-4 border-b px-3 py-1",
+      "bg-background-default flex min-h-[48px] w-full shrink-0 items-center gap-4 border-b px-3 py-1 overflow-x-hidden",
```

Comment:
To get the scrolling within the header that I was looking for we will need to:

* Allow the header to grow bigger than full width (so `min-w-full`)
* Allow overflow on the x axis at the header level (so `overflow-x-auto`)